specs/AllowGoBackForward.md

```diff
@@ -0,0 +1,142 @@
+{
+    // Get webView's current settings.
+    wil::com_ptr<ICoreWebView2Settings> webView2Settings;
+    CHECK_FAILURE(m_webview->get_Settings(&webView2Settings);
+
+    BOOL allowGoForward;
+    CHECK_FAILURE(webView2Settings->get_AllowGoForward(&allowGoForward));
+    if (allowGoForward)
+    {
+        CHECK_FAILURE(webView2Settings->put_AllowGoForward(FALSE));
+    }
+    else
+    {
+        CHECK_FAILURE(webView2Settings->put_AllowGoForward(TRUE));
+    }
+}
+```
+
+## C#
+```c#
+private WebView2 _webView;
+void ToggleAllowGoBack(object target, ExecutedRoutedEventArgs e)
+{
+    var webView2Settings = _webView.CoreWebView2.Settings;
+    if (webView2Settings.AllowGoBack)
+    {
+        webView2Settings.AllowGoBack = false;
+    }
+    else
+    {
+        webView2Settings.AllowGoBack = true;
+    }
+}
+void ToggleAllowGoForward(object target, ExecutedRoutedEventArgs e)
+{
+    var webView2Settings = _webView.CoreWebView2.Settings;
+    if (webView2Settings.AllowGoForward)
+    {
+        webView2Settings.AllowGoForward = false;
+    }
+    else
+    {
+        webView2Settings.AllowGoForward = true;
+    }
+}
+```
+
+# Remarks
+
+# API Notes
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+
+## Win32 C++
+```c++
+/// A continuation of the ICoreWebView2Settings interface to manage allowing navigating
+/// back and forward.
+[uuid(d21886f5-03fd-4029-8179-d7ed9f726b06), object, pointer_default(unique)]
+interface ICoreWebView2Settings : IUnknown {
+  /// `AllowGoBack` controls whether allow user to navigate back. If it is false, we
+  /// will block all back navigations regardless of source (button click, xbutton click,
+  /// key event and etc.).
+  /// The `CanGoBack` API will also return false if we set `AllowGoBack` to disabled.
+  /// This setting changes will work in the next navigation, which means that changes made
+  /// after `NavigatingStarting` event do not apply until the next top-level navigation.
+  /// The default value is true.
+  [propget] HRESULT AllowGoBack([out, retval] BOOL* allow);
+  /// Set the `AllowGoBack` property.
+  [propput] HRESULT AllowGoBack([in] BOOL allow);
+  /// `AllowGoForward` controls whether allow user to navigate forward. If it is false, we
+  /// will block all forward navigations regardless of source (button click, xbutton click,
+  /// key event and etc.).
+  /// The `CanGoForward` API will also return false if we set `AllowGoForward` to disabled.
+  /// This setting changes will work in the next navigation, which means that changes made
+  /// after `NavigatingStarting` event do not apply until the next top-level navigation.
+  /// The default value is true.
+  [propget] HRESULT AllowGoForward([out, retval] BOOL* allow);
+  /// Set the `AllowGoForward` property.
+  [propput] HRESULT AllowGoForward([in] BOOL allow);
+}
+```
+
+## .NET and WinRT
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    public partial class CoreWebView2Settings
```

Comment:
Previously we allowed for either C# or MIDL3 but some months ago we switched to just MIDL3. You can copy paste out of the generated Microsoft.Web.WebView2.Core.IDL file after running build-apiwriter.ps1. See recent PRs for other examples.

Reply:
Thanks for explaining, updated the c# code.